apiv2/db_import/tests/populate_db.py

```diff
@@ -0,0 +1,537 @@
+from datetime import datetime
+
+import sqlalchemy as sa
+from database.models import (
+    Annotation,
+    AnnotationAuthor,
+    AnnotationFile,
+    AnnotationShape,
+    Dataset,
+    DatasetAuthor,
+    DatasetFunding,
+    Deposition,
+    DepositionAuthor,
+    Run,
+    Tiltseries,
+    Tomogram,
+    TomogramAuthor,
+    TomogramVoxelSpacing,
+)
+
+DATASET_ID = 30001
+DATASET_AUTHOR_ID = 301
+DATASET_FUNDING_ID = 302
+DEPOSITION_ID1 = 300
+DEPOSITION_ID2 = 301
+DEPOSITION_AUTHOR_ID = 201
+RUN1_ID = 401
+RUN4_ID = 402
+TILTSERIES_ID = 501
+TOMOGRAM_VOXEL_ID1 = 502
+TOMOGRAM_VOXEL_ID2 = 503
+TOMOGRAM_ID = 601
+TOMOGRAM_AUTHOR_ID = 701
+ANNOTATION_ID = 602
+ANNOTATION_FILE_ID = 701
+ANNOTATION_AUTHOR_ID = 702
+
+STALE_RUN_ID = 902
+STALE_TOMOGRAM_ID = 903
+STALE_ANNOTATION_ID = 904
+
+
+def model_to_dict(sa_object):
+    return {item.key: getattr(sa_object, item.key) for item in sa.inspect(sa_object).mapper.column_attrs}
+
+
+def stale_deposition_metadata() -> dict:
+    return {
+        "id": DEPOSITION_ID1,
+        "deposition_title": "Test Deposition",
+        "deposition_description": "Test Description",
+        "deposition_date": datetime.now().date(),
+        "release_date": datetime.now().date(),
+        "last_modified_date": datetime.now().date(),
+        "publications": "Publications",
+    }
+
+
+def stale_deposition_author() -> dict:
+    return {
+        "id": DEPOSITION_AUTHOR_ID,
+        "deposition_id": DEPOSITION_ID1,
+        "name": "Author 1",
+        "author_list_order": 1,
+        "corresponding_author_status": False,
+        "primary_author_status": False,
+    }
+
+
+def write_data(function):
+    def wrapper(session: sa.orm.Session, *args, **kwargs):
+        obj = function(session, *args, **kwargs)
+        if obj:
+            session.add(obj)
+        session.flush()
+
+    return wrapper
+
+
+@write_data
+def populate_deposition(session: sa.orm.Session) -> Deposition:
+    return Deposition(**stale_deposition_metadata())
+
+
+@write_data
+def populate_deposition_authors(session: sa.orm.Session) -> DepositionAuthor:
+    return DepositionAuthor(**stale_deposition_author())
+
+
+@write_data
+def populate_dataset(session: sa.orm.Session) -> Dataset:
```

Comment:
TODO: Make sure that the `Dataset.deposition_id` and `Tomogram.deposition_id` fields are not nullable, and update the tests to create depositions automatically when we create datasets, so that we do *not* need to run these `populate_xxxx` methods in a special order.